src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -886,20 +891,26 @@ private static void RestoreViewName(DataSourceEntry ds, string dataSetName)
             var text = PAWriterVisitor.PrettyPrint(ir);
             dir.WriteAllText(subDir, filename, text);
 
-            var extraData = new Dictionary<string, ControlState>();
+            // The key isn't used during deserialization, so just give the
+            // parent object a unique key so it can be retrieved when loaded.
+            var controls = new Dictionary<string, ControlState>();
             foreach (var item in app._editorStateStore.GetControlsWithTopParent(topParentname ?? controlName))
             {
-                extraData.Add(item.Name, item);
+                if (item.IsTopParent ?? false)
+                    controls.Add(TopParentObjectKey, item);
+                else
+                    // Preface with an underscore to prevent unlikely collision with top parent
+                    controls.Add($"_{item.Name}", item);
```

Comment:
This seems a bit hacky. We shouldn't be putting names into unpacked files that aren't obviously meaningful if we can avoid it.